vignettes/epidist.Rmd

```diff
@@ -1,201 +1,211 @@
+Finally, in reality, it's not possible to observe every case. We suppose that a random sample of individuals of size `sample_size` are observed:
 
 ```{r}
-latent_truncation_censoring_fit <- latent_truncation_censoring_adjusted_delay(
-  data = truncated_obs, cores = 4, refresh = 0
-)
+sample_size <- 200
 ```
 
-### Summarise model posterior and compare to known truth
-
-Combine models into a named list.
+This sample size corresponds to `r 100 * round(sample_size / nrow(obs_cens_trunc), 3)`% of the data.
 
 ```{r}
-models <- list(
-  "Latent variable truncation and censoring adjusted" = latent_truncation_censoring_fit
-)
+obs_cens_trunc_samp <-
+  obs_cens_trunc[sample(seq_len(.N), sample_size, replace = FALSE)]
 ```
 
-Extract and summarise lognormal posterior estimates.
+With our censored, truncated, and sampled data, we are now ready to attempt to recover the underlying delay distribution using `epidist`.
+
+# Fit the model {#fit}
 
-```{r lognormal-draws}
-draws <- models |>
-  map(extract_lognormal_draws) |>
-  rbindlist(idcol = "model") |>
-  rbind(epinowcast_draws, use.names = TRUE)
+If we had access to `obs`, then it would be simple to estimate the delay distribution.
+However, with only access to `obs_cens_trunc_samp`, we must use a statistical model to avoid severely biased estimates!
 
-draws <- draws[,
-   model := factor(
-    model, levels = c("Joint incidence and forward delay", rev(names(models)))
-   )]
+The main model function in `epidist` is `latent_truncation_censoring_adjusted_delay`.
+<!-- In a future vignette, we will explain in more detail the structure of the model. -->
+The parameters of this model are inferred using Bayesian inference.
+In particular, we use the the No-U-Turn Sampler (NUTS) Markov chain Monte Carlo (MCMC) algorithm via the `brms` R package.
 
-summarised_draws <- draws |>
-  draws_to_long() |>
-  summarise_draws(sf = 3)
+```{r}
+fit <- latent_truncation_censoring_adjusted_delay(
+  data = obs_cens_trunc_samp, cores = 4, refresh = 0
+)
 
-knitr::kable(summarised_draws[parameter %in% c("meanlog", "sdlog")])
+summary(fit)
 ```
 
-Plot summarised posterior estimates from each model compared to the ground truth.
+# Compare estimates {#compare}
 
-```{r, fig.width = 9, fig.height = 4}
-draws |>
-  draws_to_long() |>
-  make_relative_to_truth(draws_to_long(secondary_dist)) |>
-  plot_relative_recovery(y = model, fill = model) +
-  facet_wrap(vars(parameter), nrow = 1, scales = "free_x") +
-  scale_fill_brewer(palette = "Dark2") +
-  guides(fill = guide_none()) +
-  labs(
-    y = "Model", x = "Relative to ground truth"
-  )
-```
+Compare modelled estimate to underlying truth here to convince user that it works.
+Perhaps as an exercise, we could show the "really simple" estimate being wrong here otherwise it won't be so impressive that the model is right.
+This section should also show the user how to get objects that they might want out of the fitted object.
```

Comment:
I think this could be nice, but raises questions about whether we should keep the "naive" model or just code it using brms in the example.

We should also choose a pretty short example delay interval if we want the adjusted estimate to be much better than the raw one.

Reply:
I think we should code it using `brms` in the example. Perhaps you could say it's the wrong place, but if someone is familiar with `y ~ x` synatx it might reinforce to them that our model is really just some fancy version of that.

Good point about choosing the simulation settings to make sure there is a difference. When I get to this I'll choose the simulation settings to demonstrate the value of using `epidist`.

Reply:
> We should also choose a pretty short example delay interval

Or long distribution?

> I think we should code it using brms in the example.

I think I agree but we may look at this and change our mind if it seems a bit complicated

Reply:
See above comment about maybe doing this via visualing the true dist and the truncated and censored dist. I think do that and then see if we need this? We might still as we can clearly show the summary parameters of the fits vs true (like the old readme but simpler).

Reply:
I think I've done this.